### PR TITLE
O3DE Blender Scene Exporter v1.4

### PR DESCRIPTION
**Pull request by shawstar**
## What does this PR do?

Features added:
**Add custom Project Path:** This feature you are able to add a custom path to save in your plugin, this path will be added to the O3DE Projects Menu Pull-down.
**User Defined Properties (UDP)** : You now are able to easily add PhysX and LOD levels to your models.
**Animation Options now support:** Skin Attachment Mesh with Rig for exporting Skin Attachments that work with O3DE.

**Fixed Issues:** Fixed export multi selected meshes from exporting incorrectly. (Each export had added selections)

_Please describe any testing performed._
Tested in Blender 3.0 and 3.2
image

### CODE REVIEW

**Review by HogJonny-AMZN — APPROVED** [review]

File: Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/Tools/DCC/Blender/AddOns/SceneExporter/ui.py [hunk 1]
```diff
@@ -46,6 +108,72 @@ def execute(self, context):
...
+    def execute(self, context):
+        """!
+        This function will create the LOD mesh
+        """
+        # Create a Pop-Up confirm window
+        bpy.types.Scene.pop_up_confirm_label = 'Adding UDP Type: o3de_atom_lod to mesh.'
+        bpy.types.Scene.pop_up_question_label = 'Add UPD to a Duplicate mesh?'
+        bpy.types.Scene.udp_type = constants.UPD.get('o3de_atom_lod')
+        bpy.ops.message_confirm.popup('INVOKE_DEFAULT')
+        return{'FINISHED'}
```
Comment (benblack75) [inline Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/Tools/DCC/Blender/AddOns/SceneExporter/ui.py:175]: I'm wondering if this is a convention I'm unaware of with Blender Python, but I'm curious why you return what looks to be a dictionary with a key and no value assigned? Couldn't this be just a string? Or does the caller require a dictionary formatted return?

File: Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/Tools/DCC/Blender/AddOns/SceneExporter/utils.py [hunk 2]
```diff
@@ -127,23 +157,40 @@ def loop_through_selected_materials(texture_file_path):
                         bpy.data.images[img.image.name].filepath = str(Path(texture_file_path).joinpath(build_string))
                     
                     bpy.data.images[img.image.name].save()
-                full_path = bpy.path.abspath(img.image.filepath, library=img.image.library)
+                full_path = Path(bpy.path.abspath(img.image.filepath, library=img.image.library))
                 base_name = Path(full_path).name
+                if not full_path.exists(): # if embedded path doesnt exist, check current folder
+                    #full_path = str(Path(bpy.data.filepath).parent.joinpath(base_name))
```
Comment (benblack75) [inline Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/Tools/DCC/Blender/AddOns/SceneExporter/utils.py:163]: Is there a reason for leaving this in when it is inactive?

**Review by benblack75 — APPROVED** [review]
Looks good. Just left a couple comments, but nothing critical. I would definitely be interested in knowing why you are returning your 'FINISHED' messaging as a dict with a single key and no value? I've never seen that before

File: Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/Tools/DCC/Blender/AddOns/SceneExporter/ui.py (hunk 1 above)
Comment (shawstar) [reply Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/Tools/DCC/Blender/AddOns/SceneExporter/ui.py:175]: Yeah, this is just a value string that is called, I sort of set it up the groundwork to in the future have even more added to this dictionary that just 1 default variable, like a bunch of defaults for the UDP. But, not just on this iteration.

File: Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/Tools/DCC/Blender/AddOns/SceneExporter/utils.py (hunk 2 above)
Comment (shawstar) [reply Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/Tools/DCC/Blender/AddOns/SceneExporter/utils.py:163]: Opps, I will take this out. Thanks.

**Pull request closed without merging**